### PR TITLE
[improve][broker] Make cluster metadata init command support metadata config path

### PR DESCRIPTION
**Pull request by Demogorgon314**
### Motivation

The pulsar configuration supported `metadataStoreConfigPath` and `configurationStoreConfigPath`, but the cluster metadata init command does not have a place to set the config path.

### Modifications

Make cluster metadata init command support metadata config path

### Documentation

- [ ] `doc` 
- [x] `doc-required` 
- [ ] `doc-not-needed` 
- [ ] `doc-complete`

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/PulsarClusterMetadataSetup.java [hunk 1]
```diff
@@ -464,9 +476,12 @@ static void createPartitionedTopic(MetadataStore configStore, TopicName topicNam
         }
     }
 
-    public static MetadataStoreExtended initLocalMetadataStore(String connection, int sessionTimeout) throws Exception {
```
Comment (lhotari) [inline pulsar-broker/src/main/java/org/apache/pulsar/PulsarClusterMetadataSetup.java:467]: Is it useful to keep backwards compatibility by keeping this method signature and delegating to the new method signature with `configPath` set to `null`? Just thinking of possible cases where this class is used by some external code.

File: pulsar-broker/src/main/java/org/apache/pulsar/PulsarClusterMetadataSetup.java [hunk 2]
```diff
@@ -475,10 +490,13 @@ public static MetadataStoreExtended initLocalMetadataStore(String connection, in
         return store;
     }
 
-    public static MetadataStoreExtended initConfigMetadataStore(String connection, int sessionTimeout)
```
Comment (lhotari) [inline pulsar-broker/src/main/java/org/apache/pulsar/PulsarClusterMetadataSetup.java:478]: Is it useful to keep backwards compatibility by keeping this method signature and delegating to the new method signature with `configPath` set to `null`? Just thinking of possible cases where this class is used by some external code.

**Review by lhotari — APPROVED** [review]
LGTM, left comments about keeping binary compatibility

File: pulsar-broker/src/main/java/org/apache/pulsar/PulsarClusterMetadataSetup.java (hunk 1 above)
Comment (Demogorgon314) [reply pulsar-broker/src/main/java/org/apache/pulsar/PulsarClusterMetadataSetup.java:467]: Good point, updated.

**Review by lhotari — APPROVED** [review]
LGTM

**Review by Coselding — APPROVED** [review]

**Pull request merged**